### PR TITLE
Make explicit FACET and FACET_VERSION fields for our facets

### PR DESCRIPTION
**Pull request by briandealwis**
Follow pattern of WebFacetUtil and JavaFacet in creating explicit fields to hold onto the IProjectFacet and IProjectFacetVersion instances for our AppEngineStandardFacet and AppEngineFlexFacet facets.

Get rid of FacetedProjectHelper.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineFlexFacet.java [hunk 1]
```diff
@@ -29,14 +29,18 @@
   public static final String ID = "com.google.cloud.tools.eclipse.appengine.facets.flex";
   public static final String VERSION = "1";
 
+  public static final IProjectFacet FACET = ProjectFacetsManager.getProjectFacet(ID);
+  public static final IProjectFacetVersion FACET_VERSION = FACET.getVersion(VERSION);
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineFlexFacet.java]: Here it's `VERSION` whereas in `AppEngineStandardFacet`, it's `.getDefaultVersion()`.

File: plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineFlexFacet.java [hunk 2]
```diff
@@ -29,14 +29,18 @@
...
+
   /**
    * Returns true if project has the App Engine Flex facet and false otherwise.
    *
    * @param project should not be null
    * @return true if project has the App Engine Flex facet and false otherwise
    */
   public static boolean hasFacet(IFacetedProject project) {
-    return FacetedProjectHelper.projectHasFacet(project, ID);
+    Preconditions.checkNotNull(project);
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineFlexFacet.java]: Nit: the std counterpart doesn't have this.

File: plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/NewNativeAppEngineStandardProjectTest.java [hunk 3]
```diff
@@ -71,7 +70,7 @@ private void createAndCheck(String projectName, String packageName, String[] pro
     IFacetedProject facetedProject = ProjectFacetsManager.create(project);
     assertNotNull("Native App Engine projects should be faceted", facetedProject);
     assertTrue("Project does not have standard facet",
-        FacetedProjectHelper.projectHasFacet(facetedProject, AppEngineStandardFacet.ID));
+        AppEngineStandardFacet.hasFacet(facetedProject));
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/NewNativeAppEngineStandardProjectTest.java]: Nit: probably can fit in a line.

File: plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineFlexFacet.java (hunk 1 above)
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineFlexFacet.java]: Oops, good catch.  Meant to change that.

File: plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/NewNativeAppEngineStandardProjectTest.java (hunk 3 above)
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/NewNativeAppEngineStandardProjectTest.java]: I tried: 105 characters :-(

**Pull request merged**